### ISSUE TITLE
How to convert iknet quaternion output back to original Mano axis angle representation

### ISSUE THREAD

**Issue by pablovela5620** [CalciferZh/minimal-hand#80]
I want to be able to use the output from IKNet with something like [manopth](https://github.com/hassony2/manopth). Converting the output from quaternions back to axis angle is no problem. But it seems like there was more modifications done. I'm not exactly sure how to get back to the original 16 axis angle representation. 

It seems like the quaternion representation is a global rotation relative to the parent joint, so I tried the following (unsuccessfully)

```python
def convert_iknet_pose(ik_quat):
    quat = torch.tensor(ik_quat).unsqueeze(0)
    axangle = quaternion_to_angle_axis(quat).numpy()[0]
    ret_axangle = [None] * MANOHandJoints.n_joints
    for j in range(MANOHandJoints.n_joints):
        parent = MANOHandJoints.parents[j]
        if parent is not None:
            ret_axangle[j] = axangle[j, :] - axangle[parent, :]

    ret_axangle[0] = axangle[0]
    ret_axangle = np.stack(ret_axangle)
    return ret_axangle[:16, :]
```
where ik_quat is the output from `theta_mano = mpii_to_mano(theta_mpii)`
I've already validated that my conversion from quaterion to angle axis is fine so its a problem with the post-processing aftewards
any help would be appreciated. Thanks!

**Comment by CalciferZh** [comment]
`ret_axangle[j] = axangle[j, :] - axangle[parent, :]`
I think if you want to compute the "relative" rotation between two axis-angles A' and B', you need to first convert them into rotation matrices A and B, and then compute the difference A * B_inv (or A_inv * B), and finally convert them back.

**Comment by pablovela5620** [comment]
Thank you, I'll give that a try

**Comment by amundra15** [comment]
Hey @pablovela5620. Any chance you have a working snippet for the above task? It would be very helpful.
